<a href="https://colab.research.google.com/github/JaimeTorio/UFV_Visualizacion/blob/main/Ejercicios%20de%20clase/%20Clase2b.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Clase 18/10/2024**

In [1]:
import pandas as pd
df = pd.read_csv('datos_ejercicio_ventas.csv')
print(df.head())


         COUNTRY        SUBBRAND  YEAR  MONTH     SCENARIO FORECAST  \
0       Portugal     Lipton (L3)  2023     12  AI_forecast  AI_P02F   
1  Great Britain     Lipton (L3)  2023     12  AI_forecast  AI_P10F   
2          Spain  Pepsi Max (L3)  2023     12  AI_forecast  AI_P09F   
3  Great Britain        7up (L3)  2024     12  AI_forecast  AI_P10F   
4        Hungary     Lipton (L3)  2023      9  AI_forecast  AI_P03F   

   FORECAST_YEAR         AMOUNT  
0         2023.0  754356.237194  
1         2023.0  560030.558029  
2         2023.0   88501.980847  
3         2023.0  363224.511516  
4         2023.0  396176.120491  


In [2]:
# 1. Dataset size
dataset_size = df.shape

# 2. Number of Actuals and Forecast (Scenario)
actuals_forecast_count = df['SCENARIO'].value_counts()

# 3. Prediction Horizons (Year/Month combinations)
prediction_horizons = df[['YEAR', 'MONTH']].drop_duplicates().shape[0]

# 4. Number of countries and products
num_countries = df['COUNTRY'].nunique()
num_products = df['SUBBRAND'].nunique()

# 5. Historical data (Actuals and Forecast)
historical_data_range = df[['YEAR', 'MONTH']].drop_duplicates().sort_values(by=['YEAR', 'MONTH'])

# 6. Distinct Forecasts
distinct_forecasts = df['FORECAST'].nunique()

{
    'Dataset size': dataset_size,
    'Actuals and Forecast count': actuals_forecast_count.to_dict(),
    'Prediction Horizons': prediction_horizons,
    'Number of Countries': num_countries,
    'Number of Products': num_products,
    'Historical Data Range': historical_data_range.head(), # Showing first few rows
    'Distinct Forecasts': distinct_forecasts
}


{'Dataset size': (18666, 8),
 'Actuals and Forecast count': {'AI_forecast': 17766, 'actual': 900},
 'Prediction Horizons': 29,
 'Number of Countries': 9,
 'Number of Products': 6,
 'Historical Data Range':      YEAR  MONTH
 34   2023      1
 397  2023      2
 118  2023      3
 9    2023      4
 57   2023      5,
 'Distinct Forecasts': 12}

1. **Tamaño del Dataset**: 18,666 filas y 8 columnas.
2. **Número de Actuals y Forecast**:
   - AI_forecast: 17,766 registros.
   - Actuals: 900 registros.
3. **Horizontes de predicción**: Hay 29 combinaciones únicas de año y mes (horizontes de predicción).
4. **Número de países**: 9 países.
5. **Número de productos**: 6 productos (submarcas).
6. **Histórico de datos (Actuals y Forecast)**: El rango de datos históricos comienza en enero de 2023.
7. **Forecasts distintos**: Hay 12 tipos de forecast distintos.
